lib/Doctrine/Migrations/Tools/Console/ConsoleLogger.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Migrations\Tools\Console;
+
+use DateTime;
+use DateTimeInterface;
+use Psr\Log\AbstractLogger;
+use Psr\Log\InvalidArgumentException;
+use Psr\Log\LogLevel;
+use Symfony\Component\Console\Output\ConsoleOutputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use function get_class;
+use function gettype;
+use function is_object;
+use function is_scalar;
+use function method_exists;
+use function sprintf;
+use function strpos;
+use function strtr;
+
+/**
+ * PSR-3 compliant console logger.
+ *
+ * @see https://www.php-fig.org/psr/psr-3/
+ */
+class ConsoleLogger extends AbstractLogger
```

Comment:
```suggestion
final class ConsoleLogger extends AbstractLogger
```

Shouldn't this also be an internal class?

Reply:
true